CHANGELOG.d/breaking_disable-constraints-in-ffi.md

```diff
@@ -0,0 +1,23 @@
+* Disable type class constraints in FFI
+
+  Previously, one could write FFI like the following:
+  ```purescript
+  foreign import foo :: forall a. Show a => a -> String
+  ```
+
+  Type class dictionaries are "magically" handled by the compiler.
+  By including them in the above FFI, one can depend on their representation.
+  Since the representation can change without notice, this may silently break
+  code.
+
+  In `v0.14.x`, a warning was emitted if these were used. Now it will fail
+  to compile. Rather, one should write something like the following
+  where the members of the type class are passed explicitly to
+  the FFI function as arguments:
+
+  ```purescript
+  foo :: forall a. Show a => a -> String
+  foo val = fooImpl show val
+
+  foreign import fooImpl :: forall a. (a -> String) -> a -> String
```

Comment:
Do we have a test that exercises this pattern?

Reply:
Could you clarify what you mean by this?

Reply:
Literally just a test that verifies that calling such a `fooImpl show val` works as we expect it to, and not, say, that `show` gets passed to `fooImpl` without being applied to the appropriate dictionary.

It's pretty unlikely that this would regress without a lot of other things breaking, but if it's going in our changelog as the recommended thing to do, I'd prefer there to be even just one test that covers it.

Reply:
Ah, that makes sense. I've just pushed a commit that I believe does that.